self-core-impl/src/main/java/com/selfxdsd/core/BaseLogin.java

```diff
@@ -0,0 +1,46 @@
+package com.selfxdsd.core;
+
+import com.selfxdsd.api.Login;
+import com.selfxdsd.api.User;
+import com.selfxdsd.api.storage.Storage;
+
+/**
+ * Base login implementation that aids concrete implementations of Login.
+ * @author criske
+ * @version $Id$
+ * @since 0.0.1
+ */
+abstract class BaseLogin implements Login {
+
+    @Override
+    public User signUp(final Storage storage) {
+        return signUp(storage, userCredentials(storage));
+    }
+
+    /**
+     * An unauthenticated user factory that has the necessary credentials
+     * for sign up.
+     * @param storage Storage context that might be used by user builders.
+     * @return An unauthenticated User.
+     */
+    protected abstract User userCredentials(final Storage storage);
```

Comment:
@criske Can you name this method to simply ``user(...)``? Also, why is it ``protected``?

Reply:
Chiar daca BaseLogin e package-private, am vrut sa-i restrang scope-ul, ea nu face parte API, e folosita doar de BaseLogin. Nu as vedea unde am putea-o folosi in afara de BaseLogin. Pe Self il interseaza doar signUp-ul din Login si atat. Dar ok, o sa-i modific accesul si o sa o redenumesc